### DISCUSSION TITLE
Connect gluetun to privare wireguard server

### DISCUSSION THREAD

**Discussion by pvanryn** [qdm12/gluetun#1237] [Q&A]
I'd like to connect gluetun to a private wireguard server I have running on Oracle Cloud. The server works fine with a conventional wireguard client, but I can't connect with gluetun. It fails when attempting to connect DNS over TLS. Is there anyway to make this work, or does it have to be a supported VPN?

> wireguard    | 2022-11-13T13:41:10Z INFO IPv6 is not supported
wireguard    | 2022-11-13T13:41:10Z INFO [routing] default route found: interface eth0, gateway 172.17.0.1 and assigned IP 172.17.0.5
wireguard    | 2022-11-13T13:41:10Z INFO [routing] adding route for 0.0.0.0/0
wireguard    | 2022-11-13T13:41:10Z INFO [firewall] setting allowed subnets...
wireguard    | 2022-11-13T13:41:10Z INFO [routing] default route found: interface eth0, gateway 172.17.0.1 and assigned IP 172.17.0.5
wireguard    | 2022-11-13T13:41:10Z INFO TUN device is not available: open /dev/net/tun: no such file or directory; creating it...
wireguard    | 2022-11-13T13:41:10Z INFO [pprof] http server listening on [::]:6060
wireguard    | 2022-11-13T13:41:10Z INFO [dns over tls] using plaintext DNS at address 1.1.1.1
wireguard    | 2022-11-13T13:41:10Z INFO [http server] http server listening on [::]:8000
wireguard    | 2022-11-13T13:41:10Z INFO [firewall] allowing VPN connection...
wireguard    | 2022-11-13T13:41:10Z INFO [healthcheck] listening on 127.0.0.1:9999
wireguard    | 2022-11-13T13:41:10Z INFO [http proxy] listening on :8888
wireguard    | 2022-11-13T13:41:10Z INFO [wireguard] Using available kernelspace implementation
wireguard    | 2022-11-13T13:41:10Z INFO [wireguard] Connecting to 132.xxx.xx.xx:51820
wireguard    | 2022-11-13T13:41:10Z INFO [wireguard] Wireguard is up
wireguard    | 2022-11-13T13:41:10Z INFO [dns over tls] downloading DNS over TLS cryptographic files
wireguard    | 2022-11-13T13:41:18Z INFO [healthcheck] program has been unhealthy for 6s: restarting VPN
wireguard    | 2022-11-13T13:41:18Z INFO [vpn] stopping
wireguard    | 2022-11-13T13:41:18Z ERROR [vpn] cannot get version information: Get "https://api.github.com/repos/qdm12/gluetun/commits": context canceled

**Comment by qdm12** [comment]
Hello, sorry for the huge delay answering, that project became quite too-popular to handle without lag 😸 

The connection is likely just not working, Wireguard won't complain since it's a "silent" protocol, unlike openvpn, it won't cry about receiving 0 byte back.

I would suggest you to check your keys 😉 

Also thanks for the tip 🤑!
